extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -70,6 +71,62 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(
+      StateProperty.PAGE_SIZE,
+      (pageSizeState) => {
+        this.flipTagIfOffscreen_(pageSizeState);
+      },
+      true /** callToInitialize */
+    );
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @param {!Object} pageSize
+   * @private
+   */
+  flipTagIfOffscreen_(pageSize) {
+    /* We only check the right hand side, as resizing only expands the border to the right. */
+
+    const storyPageWidth = pageSize.width;
+
+    this.measureElement(() => {
+      /*
+       * We are using offsetLeft and offsetWidth instead of getLayoutBox() because
+       * the correct measurements are not taken into account when using a CSS transform (such as translate),
+       * which we are using in the i-amphtml-amp-story-shopping-tag-inner-flipped class.
+       */
+      const {offsetLeft, offsetWidth} = this.element;
+
+      let shouldFlip;
+
+      this.measureMutateElement(
+        () => {
+          shouldFlip = offsetLeft + offsetWidth > storyPageWidth;
```

Comment:
Re: this is not doing any DOM reading operation so it doesn't need to be in a measureElement context. Let me know if I can help you with more documentation for these